source/iface/properties.rst

```diff
@@ -2,20 +2,191 @@
   Copyright 2020 The Khronos Group Inc.
   SPDX-License-Identifier: CC-BY-4.0
 
-.. _property_list:
+.. _properties:
 
+**********
+Properties
+**********
+
+Each of the following SYCL runtime classes: ``accessor``, ``buffer``,
+``host_accessor``, ``host_sampled_image_accessor``,
+``host_unsampled_image_accessor``, ``context``,
+``local_accessor``, ``queue``, ``sampled_image``,
+``sampled_image_accessor``, ``stream``, ``unsampled_image``,
+``unsampled_image_accessor`` and ``usm_allocator``
+provide an optional parameter in each of their constructors
+to provide a ``property_list`` which contains zero or more properties.
+Each of those properties augments the semantics of the class with
+a particular feature. Each of those classes must also provide
+``has_property`` and ``get_property`` member functions
+for querying for a particular property.
+
+The example below illustrates how using properties does not affect the
+type of the object, thus, does not prevent the usage of SYCL objects
+in containers.
+
+.. _ buffer-properties-example:
+
+``buffer-properties-example``
```

Comment:
Maybe we should move it to the end of the file? and the link 2 lines above is broken..

Reply:
removed